browser/brave_wallet/brave_wallet_service_unittest.cc

```diff
@@ -219,6 +219,11 @@ class TestBraveWalletServiceObserver
 
   void OnNetworkListChanged() override { network_list_changed_fired_ = true; }
 
+  void OnDiscoverAssetsCompleted(
+      std::vector<mojom::BlockchainTokenPtr> discovered_assets) override {
+    on_discover_assets_completed_fired_ = true;
```

Comment:
Please use `MOCK_METHOD` with `EXPECT_CALL`, you can reference `components/brave_wallet/browser/eth_block_tracker_unittest.cc` as an example.